src/Mono.Android/Android.Runtime/OutputStreamInvoker.cs

```diff
@@ -9,25 +9,26 @@ public class OutputStreamInvoker : Stream
 
 		public OutputStreamInvoker (Java.IO.OutputStream stream)
 		{
+			if (stream == null)
+				throw new ArgumentNullException (nameof (stream));
+
 			this.BaseOutputStream = stream;
 		}
 
 		public override void Close ()
 		{
-			BaseOutputStream.Close ();
+			JavaCallWrapper.Call (() => BaseOutputStream.Close (), JavaCallWrapper.MakeIOException);
 		}
 
 		protected override void Dispose (bool disposing)
 		{
-			if (disposing && BaseOutputStream != null) {
```

Comment:
This change is unsafe.  If `disposing` is false, meaning the GC is collecting this instance, then `BaseOutputStream` may have already been collected, and we *should not* call `Dispose()` upon it.